python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -355,25 +361,35 @@ def sync(
 
             yield from self._exec_sling_cmd(cmd, encoding=encoding)
 
-    @public
     def replicate(
         self,
         *,
-        replication_config: SlingReplicationParam,
-        dagster_sling_translator: DagsterSlingTranslator,
+        context: Union[OpExecutionContext, AssetExecutionContext],
+        replication_config: Optional[SlingReplicationParam] = None,
+        dagster_sling_translator: Optional[DagsterSlingTranslator] = None,
         debug: bool = False,
     ) -> Generator[MaterializeResult, None, None]:
         """Runs a Sling replication from the given replication config.
 
         Args:
             replication_config: The Sling replication config to use for the replication.
-            dagster_sling_translator: The translator to use for the replication.
+            context: Asset or Op execution context.
             debug: Whether to run the replication in debug mode.
 
         Returns:
             Optional[Generator[MaterializeResult, None, None]]: A generator of MaterializeResult
         """
-        replication_config = validate_replication(replication_config)
+        # attempt to retrieve params from asset context if not passed as a parameter
+        if not (replication_config or dagster_sling_translator):
+            metadata_by_key = context.assets_def.metadata_by_key
+            first_asset_metadata = next(iter(metadata_by_key.values()))
+            dagster_sling_translator = first_asset_metadata.get(METADATA_KEY_TRANSLATOR)
+            replication_config = first_asset_metadata.get(METADATA_KEY_REPLICATION_CONFIG)
+
+        # if translator has not been defined on metadata _or_ through param, then use the default constructor
+        dagster_sling_translator = dagster_sling_translator or DagsterSlingTranslator()
+
+        replication_config = validate_replication(replication_config or {})
```

Comment:
Just make `validate_replication` accept an optional?

```suggestion
        replication_config = validate_replication(replication_config)
```

Reply:
The `check_inst` doesn't satisfy _pyright_ to ensure replication_config is not None, causing the return statement to throw:

> Expression of type "Mapping[str, Any] | None" cannot be assigned to return type "Mapping[str, Any]"    Type "Mapping[str, Any] | None" cannot be assigned to type "Mapping[str, Any]"      "None" is incompatible with "Mapping[str

But I can add a `replication = replication or {}` in the `validate_replication` method.

Reply:
sounds good to me